### ISSUE TITLE
Service arguments are not verified against __construct()

### ISSUE THREAD

**Issue by brad8506** [mglaman/phpstan-drupal#465]
I noticed an issue where arguments to classes in the services file were not verified against the __construct at any phpstan runlevel.
Is this something that would be fixable?

![image](https://user-images.githubusercontent.com/4384967/183574500-9bf11d6d-759a-432a-9f99-f8437fde8f51.png)

**Comment by mglaman** [comment]
Correct, this is not analyzed. Not on any level of PHPStan, it needs a custom rule. This probably exists in phpstan-symfony, but that package reads the dumped container information.

**Comment by mglaman** [comment]
I believe PHPStan 1.8.0 made this feasible by using Collectors: https://phpstan.org/developing-extensions/collectors

This would collect on `__construct` for `getNodeType`. Then in `processNode` it would check the service map to see if the class is a service.

Then a custom rule could be applied to check the service definition against the arguments.
